src/Microsoft.Diagnostics.Runtime/src/DataReaders/MacOS/MacOSProcessDataReader.cs

```diff
@@ -231,20 +239,34 @@ public unsafe bool GetThreadContext(uint threadID, uint contextFlags, Span<byte>
             if (!_threadActs.TryGetValue(threadID, out var threadAct))
                 return false;
 
-            int regSize = sizeof(x86_thread_state64_t);
+            int stateFlavor;
+            int stateCount;
+            int regSize;
+            (stateFlavor, stateCount, regSize) = Architecture switch
+            {
+                Architecture.Arm64 => (Native.ARM_THREAD_STATE64, Native.ARM_THREAD_STATE64_COUNT, sizeof(arm_thread_state64_t)),
+                _ => (Native.x86_THREAD_STATE64, Native.x86_THREAD_STATE64_COUNT, sizeof(x86_thread_state64_t))
+            };
 
             byte[] buffer = ArrayPool<byte>.Shared.Rent(regSize);
             try
             {
                 fixed (byte* data = buffer)
                 {
-                    int stateCount = Native.x86_THREAD_STATE64_COUNT;
-                    int kr = Native.thread_get_state(threadAct, Native.x86_THREAD_STATE64, new IntPtr(data), ref stateCount);
+                    int kr = Native.thread_get_state(threadAct, stateFlavor, new IntPtr(data), ref stateCount);
                     if (kr != 0)
                         return false;
                 }
 
-                Unsafe.As<byte, x86_thread_state64_t>(ref MemoryMarshal.GetReference(buffer.AsSpan())).CopyContext(context);
+                switch (Architecture)
+                {
+                    case Architecture.Arm64:
+                        Unsafe.As<byte, arm_thread_state64_t>(ref MemoryMarshal.GetReference(buffer.AsSpan())).CopyContext(context);
+                        break;
+                    default:
```

Comment:
This should be `case Architecture.X86`.  In the future if more architectures are added and supported for OS X, it makes the code less error prone.